### DISCUSSION TITLE
how to genereate GUID (ID) to use in SQL-INSERT INTO Statement

### DISCUSSION THREAD

**Discussion by Gru0810** [advplyr/audiobookshelf#2087] [Q&A]
I create long time ago an MS Access DB/App as Master to manage all my Audiobooks.
After upgrade ABS to 2.4.01 I was able to connect easely to the SQLite-DB. I try to create INSERT INTO -SQL-State to fillup mediaProgresses with all my Info in my Master-DB. I know that I can do it by API  but for me this way is easier.
My Question ist : How do you create the ID ?
I found this with Google:
   lower(
       hex(randomblob(4)) || '-' || 
       hex(randomblob(2)) || '-' || 
       '4' ||  substr(hex( randomblob(2)), 2) || '-' || 
       substr('AB89', 1 + (abs(random()) % 4) , 1)  ||  substr(hex(randomblob(2)), 2) || '-' ||
       hex(randomblob(6)
   )
Is this the way you do it?

**Comment by WarDrake** [comment]
I don't know how ABS does it, but I had cause to play around with this today, here's how I did it

substr(hex(randomblob(4)), 1, 8) || '-' ||
substr(hex(randomblob(2)), 1, 4) || '-' ||
'4' || substr(hex(randomblob(2)), 1, 3) || '-' ||
'a' || substr(hex(randomblob(2)), 1, 3) || '-' ||
substr(hex(randomblob(6)), 1, 12) AS id,
